src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateSolicitorDetailsService.java

```diff
@@ -52,4 +68,50 @@ private Map<String, Object> convertOrganisationAddressToSolicitorAddress(Organis
             .postCode(organisationData.getContactInformation().get(0).getPostcode())
             .build(), Map.class);
     }
+
+    public Map<String, Object> updateSolicitorContactDetails(ChangedRepresentative addedSolicitor,
+                                                             Map<String, Object> caseData,
+                                                             boolean isConsented,
+                                                             boolean isApplicant) {
+        if (isApplicant) {
+            caseData.put(isConsented ? CONSENTED_SOLICITOR_NAME : CONTESTED_SOLICITOR_NAME,
```

Comment:
should extract this section as well as the Respondent section was extracted to keep it concise